lib/geordi/settings.rb

```diff
@@ -1,83 +1,153 @@
-class GeordiSettings
-  require 'yaml'
+module Geordi
+  class Settings
+    require 'yaml'
 
-  GLOBAL_SETTINGS_FILE_NAME = '.config/geordi/global.yml'.freeze
-  LOCAL_SETTINGS_FILE_NAME = '.geordi.yml'.freeze
+    require File.expand_path('util', __dir__)
 
-  ALLOWED_GLOBAL_SETTINGS = %w[ pivotal_tracker_api_key ].freeze
-  ALLOWED_LOCAL_SETTINGS = %w[ headless_browser pivotal_tracker_project_id ].freeze
+    GLOBAL_SETTINGS_FILE_NAME = Util.testing? ? './tmp/global_settings.yml'.freeze : File.join(ENV['HOME'], '.config/geordi/global.yml').freeze
+    LOCAL_SETTINGS_FILE_NAME = Util.testing? ? './tmp/local_settings.yml'.freeze : './.geordi.yml'.freeze
 
-  def initialize
-    read_settings
-  end
+    GITPT_FILE_NAME = '.gitpt'.freeze
+    PROJECT_IDS_FILE_NAME = '.pt_project_id'.freeze
 
-  # Global settings
-  def pivotal_tracker_api_key
-    @global_settings['pivotal_tracker_api_key']
-  end
+    ALLOWED_GLOBAL_SETTINGS = %w[ pivotal_tracker_api_key ].freeze
+    ALLOWED_LOCAL_SETTINGS = %w[ use_vnc pivotal_tracker_project_ids ].freeze
 
-  def pivotal_tracker_api_key=(value)
-    @global_settings['pivotal_tracker_api_key'] = value
-    save_global_settings
-  end
+    def initialize
+      read_settings
+    end
 
-  # Local settings
-  # They should not be changed by geordi to avoid unexpected diffs, therefore
-  # there are no setters for these settings
-  def headless_browser
-    @local_settings['headless_browser']
-  end
+    # Global settings
+    def pivotal_tracker_api_key
+      @global_settings['pivotal_tracker_api_key'] || gitpt_api_key_old || inquire_pt_api_key
+    end
 
-  def pivotal_tracker_project_id
-    @local_settings['pivotal_tracker_project_id']
-  end
+    def pivotal_tracker_api_key=(value)
+      @global_settings['pivotal_tracker_api_key'] = value
+      save_global_settings
+    end
 
-  private
+    # Local settings
+    # They should not be changed by geordi to avoid unexpected diffs, therefore
+    # there are no setters for these settings
+    def use_vnc?
+      @local_settings.fetch('use_vnc', true)
+    end
 
-  def read_settings
-    global_path = File.join(ENV['HOME'], GLOBAL_SETTINGS_FILE_NAME)
-    local_path = LOCAL_SETTINGS_FILE_NAME
+    def pivotal_tracker_project_ids
+      project_ids = @local_settings['pivotal_tracker_project_ids'] || pt_project_ids_old
 
-    if File.exists?(global_path)
-      global_settings = YAML.safe_load(File.read(global_path))
-      check_for_invalid_keys(global_settings, ALLOWED_GLOBAL_SETTINGS, global_path)
+      case project_ids
+      when Array
+        return project_ids if project_ids.size > 0
+      when String
+        return project_ids.split(/[\s]+/).map(&:to_i)
+      when Integer
+        return [project_ids]
+      end
```

Comment:
The first `when` will return `nil` on empty array. This will crash `Geordi::Gitpt#load_projects` https://github.com/makandra/geordi/pull/100/files#diff-8e55a5101bab5a11fa81cb519817f640R37

Please do the normalization to array as you're doing now (except for that `if` in L42), but avoid the early returns. Instead, add an `if` block below the `case` that
- returns available project ids or
- exits with the warning below when empty